compiler/rustc_trait_selection/src/traits/query/normalize.rs

```diff
@@ -65,6 +67,14 @@ impl<'cx, 'tcx> AtExt<'tcx> for At<'cx, 'tcx> {
             universes: vec![],
         };
 
+        if value.has_escaping_bound_vars() {
```

Comment:
Doing it this way fixes #87219

Reply:
I think this could use a comment, ideally with an example. I think I understand what it's doing, though -- this is kind of a "consequence" of sorts of the lazy fallback, right? We're saying that if we have escaping bound vars, those are universes that we postponed creating, so we have to extend the set of universes here semi-artificially?

(If those were placeholders, they would be handled by default, right?)

If so, maybe we want to .. do the instantiation eagerly here instead? Is that something we can easily do?

Reply:
No, so this is actually a consequence by the way `normalize_erasing_regions` works. Because it needs to call the `normalize_generic_arg_after_erasing_regions`, it folds through tys and consts in a `TypeFoldable`. Importantly, it skips binders, leaving us with trying to normalize with escaping bound vars.

Here, we just add the universes that we *would* have created had we passed through the binders.

We *could* instantiate eagerly here, but it doesn't seem really necessary to me. The rest of the code is already set up to be lazy about replacing bound vars and only when we actually have to normalize.